### ISSUE TITLE
Issue loading Geyser plugin using `solana-test-validator`

### ISSUE THREAD

**Issue by Eliascm17** [solana-labs/solana#24600]
#### Problem

Machine: `MacOS`
OS Version: `12.3.1`
Solana Version: `1.10.8`
Rust Version: `1.60.0`
repo: [Cronos](https://github.com/cronos-so/cronos)

Currently developing a plugin and having issues loading it into a test validator instance. I have tried a couple of things to see how I could get around this error but still not able to find a fix. I keep getting the error `illegal hardware instruction` when I run `solana-test-validator -r --geyser-plugin-config config.json`. When I first encountered this problem I did everything to try to resolve this including making our plugin run as "vanilla" as possible, update solana versions, update solana dependencies, update rust versions, and nothing seems to have fixed it. I even went as far as containerizing our code to see if it was just my local dev environment and with linux I got a similar but different error (probably because of the difference in the machine contexts) `segmentation fault`. The last thing I tried was testing [Blockdaemon's](https://github.com/Blockdaemon/solana-accountsdb-plugin-kafka) (@terorie) plugin implementation and still no luck. 

**Note**: I did recently update my Macbook OS version (I deeply regret doing so) about a week ago and aligns with how long I've been having this issue, so this could definitely be the reason but still not sure how to go about this fix.  

Images: 
<img width="948" alt="Screen Shot 2022-04-21 at 2 14 35 PM" src="https://user-images.githubusercontent.com/20745708/164776525-d9ba2ce6-9de4-4a6b-bef3-decdd8fe0037.png">
<img width="869" alt="Screen Shot 2022-04-22 at 1 48 34 PM" src="https://user-images.githubusercontent.com/20745708/164776530-c1bb7f35-84f5-45eb-954c-39309b420e0e.png">

#### Proposed Solution
- Possible dev environment issue. Not sure what though

**Comment by Eliascm17** [comment]
**update:** just reinstalled rust 1.59.0 with `rustup default 1.59.0` and the issue is persisting

**Comment by khaIilnafis** [comment]
Can you try running the test-validator with the --no-bpf-jit arg? Error is common on M1 machines without this arg

solana-test-validator --no-bpf-jit

**Comment by Eliascm17** [comment]
> Can you try running the test-validator with the --no-bpf-jit arg? Error is common on M1 machines without this arg
> 
> solana-test-validator --no-bpf-jit

Yeah I forgot to mention that I am not using an M1 Mac. 2019 Intel based Macbook Pro. I tried that flag anyways and still getting that error.

**Comment by Eliascm17** [comment]
**Solution:** We have a forked version of the [anchor](https://github.com/project-serum/anchor) repo with the latest solana testnet versions in order to stay ahead developmentally, and at the moment that repo references `v1.10.9`.Once we rolled back to `v1.10.8` we were able to fix the error that we were getting. We'll make sure not to get too ahead of ourselves when trying to develop for testnet.
